src/algorithm/good_thomas_algorithm.rs

```diff
@@ -9,26 +11,22 @@ use algorithm::FFTAlgorithm;
 pub struct GoodThomasAlgorithm<T> {
     width: usize,
     // width_inverse: usize,
-    width_size_fft: Box<FFTAlgorithm<T>>,
+    width_size_fft: Rc<FFTAlgorithm<T>>,
 
     height: usize,
     // height_inverse: usize,
-    height_size_fft: Box<FFTAlgorithm<T>>,
+    height_size_fft: Rc<FFTAlgorithm<T>>,
 
     input_map: Vec<usize>,
     output_map: Vec<usize>,
-
-    scratch: Vec<Complex<T>>,
 }
 
-impl<T> GoodThomasAlgorithm<T>
-    where T: Signed + FromPrimitive + Copy
-{
-    pub fn new(n1: usize,
-               n1_fft: Box<FFTAlgorithm<T>>,
-               n2: usize,
-               n2_fft: Box<FFTAlgorithm<T>>)
-               -> Self {
+impl<T: FFTnum> GoodThomasAlgorithm<T> {
+    #[allow(dead_code)]
```

Comment:
I think that since this library supports so many FFT algorithms, it would be good to expose all of the different algorithms to the user in addition to providing the planner. So I think it's fine to keep this in for now. Eventually I'd like to be able to let the user decide to use any of the provided algorithms if they want, including this one.